src/DynamoCore/Graph/Nodes/CustomNodes/CustomNodeController.cs

```diff
@@ -16,23 +17,53 @@ public class CustomNodeController<T> : FunctionCallNodeController<T>
         
         protected override void InitializeInputs(NodeModel model)
         {
-            model.InPortData.Clear();
-
             if (Definition.DisplayParameters == null || Definition.Parameters == null)
                 return;
 
-            var inputs = Definition.DisplayParameters.Zip(Definition.Parameters, (dp, p) => Tuple.Create(dp, p.Description, p.DefaultValue));
-            foreach (var p in inputs)
-                model.InPortData.Add(new PortData(p.Item1, p.Item2, p.Item3));
+            var inputs = Definition.DisplayParameters.Zip(Definition.Parameters, (dp, p) => Tuple.Create(dp, p.Description, p.DefaultValue)).ToList();
```

Comment:
The following logic originally lived on `NodeModel` in `RegisterOutputs()`. We move it here for custom nodes so that their ports are not deleted when the underlying function definition changes, but the ports are updated.